components/org.wso2.carbon.identity.oauth.dcr/src/main/java/org/wso2/carbon/identity/oauth/dcr/service/DCRMService.java

```diff
@@ -694,4 +709,17 @@ private void validateRequestTenantDomain(String clientId) throws DCRMException {
                     .getMessage(), clientId));
         }
     }
+
+    /**
+     * Create a deep copy of the input Service Provider.
+     *
+     * @param serviceProvider Service Provider.
+     * @return Clone of serviceProvider.
+     */
+    private ServiceProvider createSPClone(ServiceProvider serviceProvider) {
```

Comment:
better to change the method name to cloneServiceProvider, which gives a proper meaning of the method

Reply:
Fixed